### PR TITLE
Fix metadata and pickup updated vectormath headers.

### PR DESCRIPTION
**Pull request by ashtat**
- Incorrect metadata on plugin binaries was causing load failures on Android
- Picked up new VectorMath headers with a Neon fix that addresses buzzing on Android

### CODE REVIEW

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta [hunk 1]
```diff
@@ -71,7 +71,7 @@ PluginImporter:
     second:
       enabled: 0
       settings:
-        CPU: X86
+        CPU: ARM
```
Comment (MikeChemi) [inline Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta:74]: Shouldn't this one be ARM64?

Comment (MikeChemi) [reply Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta:74]: Oh, wait. This change was to WSA, not to Android. OK, I get it.

Comment (ashtat) [reply Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta:74]: Gah, yep, I triple-checked but somehow this got messed up, fixing it.

File: Source/include/AlignedAllocator.h [hunk 2]
```diff
@@ -23,6 +25,8 @@ namespace AlignedStore
         return ptr;
 #elif (ANDROID)
         return memalign(alignment, size);
+#elif defined(__XTENSA__)
```
Comment (MikeChemi) [inline Source/include/AlignedAllocator.h:28]: Surprised to see XTENSA referenced here.

**Review by MikeChemi — APPROVED** [review]

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta (hunk 1 above)
Comment (ashtat) [reply Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libAudioPluginMicrosoftSpatializerCrossPlatform.so.meta:74]: Ah yeah, whew, not an issue.

**Comment by NoelCross** [comment]
>   isPreloaded: 0

I think that we want the isPreloaded flag set to 1 for these plugins

---
Refers to: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libHrtfDsp.so.meta:9 in 508dc91. [](commit_id = 508dc915caba6eea2644fe175cef832ec22bb236, deletion_comment = False)

**Comment by ashtat** [comment]
> > isPreloaded: 0
> 
> I think that we want the isPreloaded flag set to 1 for these plugins
> 
> Refers to: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/Android/libs/arm64-v8a/libHrtfDsp.so.meta:9 in [508dc91](https://github.com/microsoft/spatialaudio-unity/commit/508dc915caba6eea2644fe175cef832ec22bb236). [](commit_id = [508dc91](https://github.com/microsoft/spatialaudio-unity/commit/508dc915caba6eea2644fe175cef832ec22bb236), deletion_comment = False)

I'm not sure we do, the plugin library will load it on-demand. At least in my testing I didn't need to have that set.

File: Source/include/AlignedAllocator.h (hunk 2 above)
Comment (NoelCross) [reply Source/include/AlignedAllocator.h:28]: We had agreement with the Xbox team to share XTENSA references with source code partners, but we should remove XTENSA references from public code

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86/HrtfDsp.dll.meta [hunk 3]
```diff
@@ -18,10 +18,10 @@ PluginImporter:
       settings:
         Exclude Android: 1
         Exclude Editor: 1
-        Exclude Linux64: 1
-        Exclude OSXUniversal: 1
-        Exclude Win: 1
-        Exclude Win64: 1
+        Exclude Linux64: 0
+        Exclude OSXUniversal: 0
```
Comment (NoelCross) [inline Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86/HrtfDsp.dll.meta:22]: Shouldn't these be excluded for x86 binaries?

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86_64/HrtfDsp.dll.meta [hunk 4]
```diff
@@ -17,11 +17,11 @@ PluginImporter:
...
       settings:
         Exclude Android: 1
-        Exclude Editor: 1
-        Exclude Linux64: 1
-        Exclude OSXUniversal: 1
-        Exclude Win: 1
-        Exclude Win64: 1
+        Exclude Editor: 0
+        Exclude Linux64: 0
+        Exclude OSXUniversal: 0
```
Comment (NoelCross) [inline Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86_64/HrtfDsp.dll.meta:22]: seems like these flavors should be excluded

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86/HrtfDsp.dll.meta (hunk 3 above)
Comment (ashtat) [reply Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86/HrtfDsp.dll.meta:22]: I edited these files inside Unity editor. Don't think if these need to be explicitly excluded, there's no editor UI to exclude them?
image

File: Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86_64/HrtfDsp.dll.meta (hunk 4 above)
Comment (ashtat) [reply Source/Plugins/CrossPlatformPluginGenerator/Assets/Microsoft.SpatialAudio.Spatializer.Unity/Plugins/x86_64/HrtfDsp.dll.meta:22]: Not sure these need to be explicitly excluded.

**Pull request merged**